dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClasspathChecker.java

```diff
@@ -92,54 +102,49 @@ public static void main(String[] arguments)
 
     CommandLine commandLine = ClasspathCheckOption.readCommandLine(arguments);
     ImmutableList<Path> inputClasspath = ClasspathCheckOption.generateInputClasspath(commandLine);
-    // TODO(suztomo): take command-line option to choose entry point classes for reachability
-    ImmutableSet<Path> entryPoints = ImmutableSet.of(inputClasspath.get(0));
 
-    boolean onlyReachable = commandLine.hasOption("r");
-    ClasspathChecker classpathChecker = create(onlyReachable, inputClasspath, entryPoints);
+    ImmutableSet.Builder<Path> entryPoints = ImmutableSet.builder();
+    if (commandLine.hasOption("a") || commandLine.hasOption('b')) {
+      // For an artifact list (or a BOM), the first elements in inputClasspath are the artifacts
+      // specified the list, followed by their dependencies.
+      int artifactCount = ClasspathCheckOption.generateArtifacts(commandLine).size();
+      // For Maven artifact list (or a BOM), entry point classes are ones in the list
+      entryPoints.addAll(inputClasspath.subList(0, artifactCount));
+    } else {
+      // For list of jar files, entry point classes are all classes in the files
+      entryPoints.addAll(inputClasspath);
+    }
+
+    ClasspathChecker classpathChecker = create(inputClasspath, entryPoints.build());
     ClasspathCheckReport report = classpathChecker.findLinkageErrors();
 
     System.out.println(report);
   }
 
-  /**
-   * Finds linkage errors in the input classpath and generates a classpath check report.
-   */
-  public ClasspathCheckReport findLinkageErrors() throws IOException {
-    ImmutableList<Path> jarPaths = classDumper.getInputClassPath();
-
-    ImmutableMap.Builder<Path, SymbolReferenceSet> jarToSymbols = ImmutableMap.builder();
-    for (Path jarPath : jarPaths) {
-      jarToSymbols.put(jarPath, ClassDumper.scanSymbolReferencesInJar(jarPath));
-    }
-
+  /** Finds linkage errors in the input classpath and generates a classpath check report. */
+  public ClasspathCheckReport findLinkageErrors() {
     // Validate linkage error of each reference
     ImmutableList.Builder<JarLinkageReport> jarLinkageReports = ImmutableList.builder();
-    for (Map.Entry<Path, SymbolReferenceSet> entry : jarToSymbols.build().entrySet()) {
+    for (Map.Entry<Path, SymbolReferenceSet> entry : jarToSymbols.entrySet()) {
```

Comment:
Consider using `Map.forEach`. It lets you avoid the whole `entry.getKey()`/`entry.getValue()` noise.

Reply:
That's cleaner. Updated.